### PR TITLE
Fix UDTC allowance calculation

### PR DESCRIPTION
**Pull request by ulope**
The UDTC allowance was calculated as a delta of required - current balance, but the `approve` call expects an absolute value. 
This lead to insufficient balance on the last `deposit` call.

Fixes: #238

### CODE REVIEW

File: scenario_player/utils/token.py [hunk 1]
```diff
@@ -399,19 +405,20 @@ def update_allowance(self) -> Union[Tuple[str, int], None]:
         required_allowance = self.config.settings.services.udc.token.balance_per_node * node_count
 
         log.debug(
```
Comment (palango) [inline scenario_player/utils/token.py]: Should you use `local_log` here as well?

Comment (ulope) [reply scenario_player/utils/token.py]: Probably yeah, I'll make an issue about it.

**Comment by Dominik1999** [comment]
Yeah!

File: scenario_player/utils/token.py [hunk 2]
```diff
@@ -399,19 +405,20 @@ def update_allowance(self) -> Union[Tuple[str, int], None]:
...
-            log.debug("allowance update call not required - sufficient allowance")
+            log.debug("UDTC allowance sufficient")
             return
 
-        log.debug("allowance update call required - insufficient allowance")
-        allow_amount = required_allowance - udt_allowance
+        log.debug("UDTC allowance insufficient, updating")
         params = {
-            "amount": allow_amount,
+            "amount": required_allowance,
```
Comment (hackaugusto) [inline scenario_player/utils/token.py]: isn't this a regression?

Comment (ulope) [reply scenario_player/utils/token.py]: No, that's the bugfix

Comment (hackaugusto) [reply scenario_player/utils/token.py]: Sorry, let me rephrase the question. What was wrong in the previous version?
